### PR TITLE
HOTFIX: fix round history bug with round 0

### PR DESCRIPTION
**Pull request by quang-le**
Polio users get 500 when adding round 0

Related JIRA tickets :  POLIO-1139

## Self proofreading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- prevent users to create datelog when adding round 0
- backend: raise error when receiving datelogs on create

## How to test

Explain how to test your PR.
If a specific config is required explain it here (dataset, account, profile, ...)

## Print screen / video

BEFORE
Screenshot 2023-08-21 at 14 53 29

AFTER 

Screenshot 2023-08-21 at 14 56 31

## Notes

CAN BE DEPLOYED AS HOTFIX

### CODE REVIEW

File: plugins/polio/serializers.py [hunk 1]
```diff
@@ -356,6 +356,9 @@ def create(self, validated_data):
         destructions = validated_data.pop("destructions", [])
         started_at = validated_data.get("started_at", None)
         ended_at = validated_data.get("ended_at", None)
+        datelogs = validated_data.get("datelogs", None)
+        if datelogs:
+            raise serializers.ValidationError({"datelogs": "Cannot have modification history for new round"})
```
Comment (madewulf) [inline plugins/polio/serializers.py:361]: How could this have happened previously? Why is this needed?

Comment (quang-le) [reply plugins/polio/serializers.py:361]: It could happen previously because the UI would let users add a `datelog` when they would create a round 0 in a campaign that has already other rounds. 

the PR fixes the problem front-end side. I just added the check in the backend for good measure

File: plugins/polio/js/src/components/Rounds/RoundDates/RoundDates.tsx [hunk 2]
```diff
@@ -36,11 +38,21 @@ export const RoundDates: FunctionComponent<Props> = ({
         values: { rounds = [] },
         initialValues,
     } = useFormikContext<Campaign>();
+    // const currentStartDate = rounds.find(
+    //     round => round.number === roundNumber,
+    // )?.started_at;
+    // const currentEndDate = rounds.find(
+    //     round => round.number === roundNumber,
+    // )?.ended_at;
```
Comment (beygorghor) [inline plugins/polio/js/src/components/Rounds/RoundDates/RoundDates.tsx]: 🗑️

File: plugins/polio/js/src/components/Rounds/RoundDates/RoundDates.tsx [hunk 3]
```diff
@@ -36,11 +38,21 @@ export const RoundDates: FunctionComponent<Props> = ({
...
     const currentStartDate = rounds?.[roundIndex]?.started_at;
     const currentEndDate = rounds?.[roundIndex]?.ended_at;
+    // For initial data, wee need to perform a find, because if we're adding round 0
+    // We'll be addinga round at the start of the rounds array which will lead to index related errors
     const hasInitialData = Boolean(
-        initialValues.rounds?.[roundIndex]?.started_at &&
-            initialValues.rounds?.[roundIndex]?.ended_at,
+        initialValues.rounds.find(round => round.number === roundNumber)
```
Comment (beygorghor) [inline plugins/polio/js/src/components/Rounds/RoundDates/RoundDates.tsx:46]: i would prefer to store round initial values in another variable

**Review by beygorghor — APPROVED** [review]
it's working but can be improved a bit

File: plugins/polio/js/src/components/Rounds/RoundDates/RoundDates.tsx (hunk 3 above)
Comment (quang-le) [reply plugins/polio/js/src/components/Rounds/RoundDates/RoundDates.tsx:46]: I'm not convinced it would be more legible. Since `initialValues` comes from formik, it's pretty transparent. Since `rounds` is already used, I would have to make `intialRounds`  or something, which is not much shorter and less obviously related to formik.
